### PR TITLE
[2.26.x] Fix printErrorMessage when executeWithSubject fails in SubjectCommands

### PR DESCRIPTION
**Pull request by emmberk**
#### What does this PR do?
- Fixes the error message printed out to the Karaf console when `executeWithSubject` throws an `Exception` in any class that implements `SubjectCommands`
- Update `catalog:range` command to only print table header row when there are no errors

Before:
```
admin@root()> catalog:range name ___ ___

#       ID                                name                       Title

null
```

After:
```
admin@root()> catalog:range name ___ ___
Could not parse date parameters.
```

#### How should this be tested?

Verify that both checked and unchecked exceptions are handled (log and karaf console output) correctly for comands that extend `SubjectCommands`. I tested with `catalog:range`. Here are the commands that I used:

|                        | run with username                                           | run without username                           |
|----------------------|-------------------------------------------------------------|------------------------------------------------|
| successful execution  | `catalog:range --user admin modified 01-23-2009 01-24-2009` | `catalog:range modified 01-23-2009 01-24-2009` |
| unchecked `Exception`  | `catalog:range --user admin modified 01-23-2009 01-23-2009` | `catalog:range modified 01-23-2009 01-23-2009` |
| checked `Exception`    | `catalog:range --user admin modified _ 01-23-2009`          | `catalog:range modified _ 01-23-2009`          |

Additionally, make sure that the table update in `RangeCommand` doesn't cause any regressions. Verify that output is still outputted correctly in the table and that the max title length is enforced.

#### Checklist:
- [n/a] Documentation Updated
- [n/a] Update / Add Threat Dragon models
- [x] Update / Add Unit Tests
- [ ] Update / Add Integration Tests

#### Notes on Review Process
Please see [Notes on Review Process](https://codice.atlassian.net/wiki/spaces/DDF/pages/71946981/Pull+Request+Guidelines) for further guidance on requirements for merging and abbreviated reviews. 

#### Review Comment Legend:
- ✏️ (Pencil) This comment is a nitpick or style suggestion, no action required for approval. This comment should provide a suggestion either as an in line code snippet or a gist. 
- ❓ (Question Mark) This comment is to gain a clearer understanding of design or code choices, clarification is required but action may not be necessary for approval.
- ❗ (Exclamation Mark) This comment is critical and requires clarification or action before approval.

### CODE REVIEW

File: catalog/core/catalog-core-commands/src/main/java/org/codice/ddf/commands/catalog/SubjectCommands.java [hunk 1]
```diff
@@ -68,46 +64,59 @@
...
       return subject.execute(this::executeWithSubject);
     } catch (ExecutionException e) {
-      LOGGER.info("Failed to run command: {}", e.getCause().getMessage(), e.getCause());
-      throw new InvocationTargetException(e.getCause());
-    } catch (IOException e) {
-      LOGGER.info("Failed to run command", e);
-      printErrorMessage("Failed to read password");
+      final Throwable failure = e.getCause();
+      final String message = failure.getMessage();
+      LOGGER.info("Failed to run command: {}", message, failure);
```
Comment (jlcsmith) [inline catalog/core/catalog-core-commands/src/main/java/org/codice/ddf/commands/catalog/SubjectCommands.java:99]: ❗ missing placeholder for logging parameter

Comment (emmberk) [reply catalog/core/catalog-core-commands/src/main/java/org/codice/ddf/commands/catalog/SubjectCommands.java:99]: Looks like is prints it correctly as is. Is there something I'm not seeing?
```
admin@root()> catalog:range --user admin name ___ ___
Password for admin: *****

#       ID                                name                       Title

Could not parse date parameters.
admin@root()> log:tail
12:30:20,670 | INFO  | min name ___ ___ | ice.ddf.commands.catalog.SubjectCommands   99 | atalog-core-commands | Failed to run command: Could not parse date parameters.
java.lang.InterruptedException: Could not parse date parameters.
	at org.codice.ddf.commands.catalog.RangeCommand.executeWithSubject(RangeCommand.java:127) ~[bundleFile:?]
	at org.apache.shiro.subject.support.SubjectCallable.doCall(SubjectCallable.java:90) ~[?:?]
	at org.apache.shiro.subject.support.SubjectCallable.call(SubjectCallable.java:83) ~[?:?]
	at org.apache.shiro.subject.support.DelegatingSubject.execute(DelegatingSubject.java:387) ~[?:?]
	at org.codice.ddf.commands.catalog.SubjectCommands.runWithUsername(SubjectCommands.java:95) [bundleFile:?]
	at org.codice.ddf.commands.catalog.SubjectCommands.execute(SubjectCommands.java:71) [bundleFile:?]
	...
```

Comment (jlcsmith) [reply catalog/core/catalog-core-commands/src/main/java/org/codice/ddf/commands/catalog/SubjectCommands.java:99]: my bad - you are logging the exception as well as the message - disregard!

**Review by jlcsmith — APPROVED** [review]

**Comment by emmberk** [comment]
I realized that checked and unchecked exceptions are handled differently. I pushed a commit to address that and add a unit test. I also pushed a low-risk commit to clean up the output of the `range` command that I was using to test this capability. I'll update the test steps on this PR to make sure that the different edge cases are heroed.

**Review by SmithJosh — APPROVED** [review]

**Comment by emmberk** [comment]
This is ready for a hero.

**Comment by SmithJosh** [comment]
Hero successful. Verified that:
* checked and unchecked exceptions are caught and the error message is displayed in the Karaf shell
* the output of the `catalog:range` command is formatted correctly and the max length of the title field is enforced. Also, the new table format is nice - it matches the Karaf shell command tables 🎉

**Comment by emmberk** [comment]
build now

**Pull request merged**